docs/modules/dev/pages/index.adoc

```diff
@@ -1,4 +1,12 @@
 = Apache James Developer Resources
 :navtitle: Developer Resources
 
-(TODO)
+This section is aimed at developers, meaning anybody who needs to interact
+with the source code in any way.
+
+The resources available in this section are:
+
+ * xref:architecture/index.adoc[The system architecture]
+ * xref:domain/index.adoc[The domain model]
+ * xref:code/index.adoc[The code base]
+
```

Comment:
I would add a section: James allow a user to register custom code for various plugins, allowing to extend the behaviour of the James server.

Is that what you meant by `External Components`?

Reply:
Well, in the recent email thread, @mbaechler mentioned that the hexagonal architecture was an extremely important concept, or at least that is how I interpreted the message. I am trying to reconcile that statement with what I see. The [document about hexagonal architecture](https://alistair.cockburn.us/hexagonal-architecture/) that I referenced refers to "External Devices" as the things that connect to the adapters.

Also, this image mentions "External Projects", which are external to the core, so I just named them "External Components".

Should I try to fit all this with the hexagonal architecture, or should I drop the idea of that architecture? My feeling is that if it is used, it should be clarified and used in a way that is not confusing.

What do you think?